source/beamline/BioXAS/BioXASBeamlineComponent.cpp

```diff
@@ -11,9 +12,42 @@ BioXASBeamlineComponent::~BioXASBeamlineComponent()
 
 }
 
+bool BioXASBeamlineComponent::stop()
+{
+	// Want to return true if all the controls are valid and those
+	// that can be stopped are stopped, false otherwise. Returns
+	// true if there are no child controls, the expectation is that
+	// this control can't be moving otherwise.
+
+	bool childrenValid = true;
+	bool childrenStopped = true;
+
+	foreach (AMControl *control, children_) {
```

Comment:
BioXASBeamlineComponent::addChildControl() will not add a null child, but there is no restriction in AMControl::addChildControl(). It could happen, technically. I've moved the checks for whether the children are valid and can be stopped to canStop(), maybe that helps to clarify?